### ISSUE TITLE
Hot restart remains stuck in Flutter Web stable channel on Windows using CTRL + S key

### ISSUE THREAD

**Issue by DevyankShaw** [flutter/flutter#80208]
After any change, hot restart remains stuck in Flutter Web(stable channel) and it's only getting fixed if I refresh from the browser after hot restart.
But when I switched to master channel it's working fine. I hope it's a bug in the stable channel which will be fixed very soon. For now, I am developing in the master channel.

**Expected results:** 
```
Performing hot restart...
Waiting for connection from debug service on Chrome...
Restarted application in 18,144ms.
```
**Actual results:** 
```
Performing hot restart...
Waiting for connection from debug service on Chrome...
```
<details>
  <summary>Logs</summary>

```
[√] Flutter (Channel stable, 2.0.4, on Microsoft Windows [Version 10.0.19041.867], locale en-IN)
    • Flutter version 2.0.4 at E:\Flutter Development\src\flutter
    • Framework revision b1395592de (9 days ago), 2021-04-01 14:25:01 -0700
    • Engine revision 2dce47073a
    • Dart version 2.12.2

[√] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at E:\Android App Development\Android_SDK
    • Platform android-30, build-tools 30.0.3
    • Java binary at: E:\Android App Development\Android\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[√] Android Studio (version 4.1.0)
    • Android Studio at E:\Android App Development\Android
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)

[√] IntelliJ IDEA Community Edition (version 2019.1)
    • IntelliJ at E:\IntelliJ IDEA Community Edition 2019.1
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart

[√] Connected device (2 available)
    • Chrome (web) • chrome • web-javascript • Google Chrome 89.0.4389.90
    • Edge (web)   • edge   • web-javascript • Microsoft Edge 89.0.774.75

• No issues found!

```

</details>

**Comment by darshankawar** [comment]
@DevyankShaw 
Can you provide a [complete minimal reproducible code sample](https://stackoverflow.com/help/minimal-reproducible-example) that we can use to replicate this locally ?

**Comment by darshankawar** [comment]
A similar issue https://github.com/flutter/flutter/issues/78570 that we weren't able to replicate.

**Comment by DevyankShaw** [comment]
```
class HomeScreen extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Center(
      child: Text('Home'),
    );
  }
}
```
Just simply changing the text is not updating after hot restart in Flutter Web stable

**Comment by darshankawar** [comment]
@DevyankShaw 
Tried your code sample using latest stable on web (flutter run -d chrome), but was able to hot restart everytime I made changes, as below:


https://user-images.githubusercontent.com/67046386/114508241-c2d5af00-9c51-11eb-84d1-cbc1e0323915.mov

<details>
<summary>flutter doctor -v</summary>

```
[✓] Flutter (Channel stable, 2.0.4, on Mac OS X 10.15.4 19E2269 darwin-x64,
    locale en-GB)
    • Flutter version 2.0.4 at /Users/dhs/documents/fluttersdk/flutter
    • Framework revision b1395592de (7 days ago), 2021-04-01 14:25:01 -0700
    • Engine revision 2dce47073a
    • Dart version 2.12.2

[!] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.3, Build version 12C33
    ! CocoaPods 1.9.3 out of date (1.10.0 is recommended).
        CocoaPods is used to retrieve the iOS and macOS platform side's plugin
        code that responds to your plugin usage on the Dart side.
        Without CocoaPods, plugins will not work on iOS or macOS.
        For more info, see https://flutter.dev/platform-plugins
      To upgrade see
      https://guides.cocoapods.org/using/getting-started.html#installation for
      instructions.

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome
[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6915495)    

[✓] VS Code (version 1.55.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.20.0

[✓] Connected device (3 available)
    • SM A260G (mobile)            • 5200763ebcfa861f • android-arm    • Android
      8.1.0 (API 27)
    • Darshan's iphone (mobile) • 21150b119064aecc249dfcfe05e259197461ce23 • ios
      • iOS 14.4.1
    • macOS (desktop)           • macos                                    •
      darwin-x64     • Mac OS X 10.15.4 19E2269 darwin-x64
    • Chrome (web)              • chrome                                   •
      web-javascript • Google Chrome 89.0.4389.114

! Doctor found issues in 1 category.

```

</details>

**Comment by DevyankShaw** [comment]
Actually, i usually run and test flutter apps using the play button in android studio by selecting the appropriate device. But I have checked using latest stable on web (flutter run -d chrome), it's not working. You can check the below snapshot after i made the changes until and unless i explicitly press the refresh button hot restart is not working. 

https://user-images.githubusercontent.com/47057254/114737188-62836200-9d64-11eb-9d7f-55a7085b03d1.mp4

<details>
  <summary>flutter doctor -v</summary>

> [√] Flutter (Channel stable, 2.0.4, on Microsoft Windows [Version 10.0.19041.867], locale en-IN)
>     • Flutter version 2.0.4 at E:\Flutter Development\src\flutter
>     • Framework revision b1395592de (13 days ago), 2021-04-01 14:25:01 -0700
>     • Engine revision 2dce47073a
>     • Dart version 2.12.2
> 
> [√] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
>     • Android SDK at E:\Android App Development\Android_SDK
>     • Platform android-30, build-tools 30.0.3
>     • Java binary at: E:\Android App Development\Android\jre\bin\java
>     • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)
>     • All Android licenses accepted.
> 
> [√] Chrome - develop for the web
>     • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe
> 
> [√] Android Studio (version 4.1.0)
>     • Android Studio at E:\Android App Development\Android
>     • Flutter plugin can be installed from:
>        https://plugins.jetbrains.com/plugin/9212-flutter
>     • Dart plugin can be installed from:
>        https://plugins.jetbrains.com/plugin/6351-dart
>     • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)
> 
> [√] IntelliJ IDEA Community Edition (version 2019.1)
>     • IntelliJ at E:\IntelliJ IDEA Community Edition 2019.1
>     • Flutter plugin can be installed from:
>        https://plugins.jetbrains.com/plugin/9212-flutter
>     • Dart plugin can be installed from:
>        https://plugins.jetbrains.com/plugin/6351-dart
> 
> [√] Connected device (2 available)
>     • Chrome (web) • chrome • web-javascript • Google Chrome 89.0.4389.128
>     • Edge (web)   • edge   • web-javascript • Microsoft Edge 89.0.774.75
> 
> • No issues found!
> 

</details>

**Comment by darshankawar** [comment]
@DevyankShaw 
So if I understand the issue correctly, instead of directly using `flutter run -d chrome` from IDE terminal, you are selecting `chrome` from device list and then clicking on play button to run the app. After that, how are you trying to hot restart ? because if you running not from terminal, then you don't get the options to use `r` to hot restart.

**Comment by DevyankShaw** [comment]
@darshankawar 
While using the play button to run the app, I use Ctrl+S for hot restart

**Comment by darshankawar** [comment]
@DevyankShaw 
It seem to be specific to Windows, because on Mac, hot restart shortcut is `command + \`, which hot restarts and updates the screen properly.

**Comment by darshankawar** [comment]
I'll try to replicate this on Windows 10 too to confirm.

**Comment by DevyankShaw** [comment]
@darshankawar Yes let me know.

**Comment by darshankawar** [comment]
Verified this on Windows 10 and was able to replicate this issue.

<img width="986" alt="Screenshot 2021-04-19 at 6 29 19 PM" src="https://user-images.githubusercontent.com/67046386/115240287-35a4c580-a13d-11eb-8989-1cfd8b885d3d.png">

Steps to replicate:

1. Select `chrome` from device list and hit Run button.
2. Make changes to the file and use CTRL + S to hot restart which remains stuck and doesn't update the UI.

```
class HomeScreen extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Center(
      child: Text('Home'),
    );
  }
}
```

**Comment by DevyankShaw** [comment]
This issue is now resolved in Flutter 2.2, so I am closing this issue
